File: lua/main4.lua
function dangerous(v1,v2) return math.floor(v1 / v2).error end -- Modified to move error here!
function foo(array, counter)
  if counter==0 then return dangerous(array[1], counter) end
  return foo(array, counter-1)
end
local array = {}; for i=1,1000 do array[i]=0 end

print(string.format("The result is %d", foo(array, 6)))

File: perl/main4.pl
use strict; use warnings;
sub dangerous { my ($v1,$v2)=@_; return int($v1 / $v2); }
sub foo { my ($ar,$c)=@_; return $c==0 ? dangerous($ar->[0], $c) : foo($ar, $c-1) }
my @array = (0) x 1000;
print "The result is ", foo(\@array, 6), "\n";

